tests/testlib/Makefile

```diff
@@ -27,4 +29,4 @@ libtests2n.so: ${OBJS}
 	${CC} ${CFLAGS} -shared ${LDFLAGS} -o libtests2n.so ${OBJS}
 
 libtests2n.dylib: ${OBJS}
-	test ! -f /usr/lib/libSystem.dylib || libtool ${LDFLAGS} -dynamic -o libtests2n.dylib ${OBJS}
+	${CC} ${ARCHFLAGS} ${LDFLAGS} -dynamiclib -o libtests2n.dylib ${OBJS}
```

Comment:
Yes, `DLEXT` is set to `.dylib` in `s2n.mk` only when `uname` is "Darwin", so this target should not be built on other platforms.